BridgeAppSDK/ORKResult+SBAExtension.swift

```diff
@@ -0,0 +1,341 @@
+            aTouchSample[kSpatialSpanMemoryTouchSampleLocationKey]    = NSStringFromCGPoint(sample.location)
+            aTouchSample[kSpatialSpanMemoryTouchSampleIsCorrectKey]   = sample.correct
+            
+            samples += [aTouchSample]
+        }
+        return  samples
+    }
+    
+    func makeTargetRectangleRecords(targetRectangles: [NSValue]) -> NSArray
+    {
+        var rectangles = [String]()
+        
+        for value: NSValue in targetRectangles {
+            let rectangle = value.CGRectValue()
+            let stringified = NSStringFromCGRect(rectangle)
+            rectangles += [stringified]
+        }
+        return  rectangles
+    }
+}
+
+class AnswerKeyAndValue: NSObject {
+    var key: String
+    var value: AnyObject?
+    
+    init(key: String, value: AnyObject?) {
+        self.key = key
+        self.value = value
+    }
+}
+
+protocol ORKQuestionResultAnswerJSON {
+    func jsonSerializedAnswer() -> (AnswerKeyAndValue)?
+}
+
+private var _timeOnlyFormatter: NSDateFormatter? = nil
+
+extension ORKQuestionResult: ORKQuestionResultAnswerJSON {
+    
+    func timeOnlyFormatter() -> NSDateFormatter {
+        if _timeOnlyFormatter == nil {
+            _timeOnlyFormatter = NSDateFormatter.init()
+            _timeOnlyFormatter!.dateFormat = "HH:mm:ss.SSS"
+            let enUSPOSIXLocale = NSLocale.init(localeIdentifier: "en_US_POSIX");
+            _timeOnlyFormatter!.locale = enUSPOSIXLocale
+        }
+        
+        return _timeOnlyFormatter!
+    }
+    override func bridgeData(stepIdentifier: String) -> ArchiveableResult? {
+        let filename = bridgifyFilename(self.identifier) + ".json"
+        return ArchiveableResult.init(result: self.jsonObject(), resultType: .Dictionary, filename: filename)
+    }
+    
+    public override func jsonObject() -> AnyObject {
+        var ChoiceQuestionResult = [String: AnyObject]()
+        ChoiceQuestionResult[QuestionResultQuestionTextKey] = self.questionText
+        ChoiceQuestionResult[QuestionResultQuestionTypeKey] = self.questionType.rawValue
+        ChoiceQuestionResult[kIdentifierKey] = self.identifier
+        ChoiceQuestionResult[kStartDateKey] = self.startDate
+        ChoiceQuestionResult[kEndDateKey] = self.endDate
+        ChoiceQuestionResult[QuestionResultUserInfoKey] = self.userInfo
+        if let answer = self.jsonSerializedAnswer() {
+            ChoiceQuestionResult[answer.key] = answer.value
+        }
+        return (ChoiceQuestionResult as NSDictionary).jsonObject()
+    }
+    
+    func jsonSerializedAnswer() -> (AnswerKeyAndValue)? {
+        let className = NSStringFromClass(self.classForCoder)
+        fatalError("jsonSerializedAnswer not implemented for \(className)")
+    }
+    
+    func jsonForAnswer(answer: AnyObject) -> AnyObject {
+        let dateTime = answer as? NSDate
+        switch self.questionType {
+        case .TimeOfDay:
+            return timeOnlyFormatter().stringFromDate(dateTime!)
```

Comment:
I have a pull request to add to this PR (confusing, I know). There is a method in it called `SBAJSONObjectForObject`. You could also use that.